client/landing/stepper/hooks/use-site-migration-transfer/index.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+import { useSiteTransferMutation } from './mutation';
+import { useSiteTransferStatusQuery } from './query';
+
+/**
+ * Hook to to initiate a site transfer and monitor its progress
+ */
+export const useSiteMigrationTransfer = ( siteId?: number ) => {
+	const {
```

Comment:
Unfortunately, we can't have conditional before hooks 😢 
This is because it relies on the code execution to identify the hook context.

There is a video explaining it [here](https://www.youtube.com/watch?v=o_7wgRHBzeA).